alignit/train.py

```diff
@@ -52,35 +46,47 @@ def main(cfg: TrainConfig):
     optimizer = Adam(net.parameters(), lr=cfg.learning_rate)
     criterion = MSELoss()
     net.train()
-
     for epoch in range(cfg.epochs):
         for batch in tqdm(train_loader, desc=f"Epoch {epoch+1}"):
             images = batch["images"]
+            depth_images_pil = batch["depth_images"]
             actions = batch["action"].to(device)
 
-            # Convert PIL Images to tensors and stack them properly
-            # images is a list of lists of PIL Images
-            batch_images = []
-            transform = transforms.Compose([transforms.ToTensor()])
+            batch_rgb_tensors = []
+            batch_depth_tensors = []
+            rgb_transform = transforms.Compose([transforms.ToTensor()])
+            depth_transform = transforms.Compose([transforms.ToTensor()])
 
-            for image_sequence in images:
-                tensor_sequence = [
-                    transform(img.convert("RGB")) for img in image_sequence
+            for i, image_sequence in enumerate(images):
+                # Process RGB
+                tensor_sequence_rgb = [
+                    rgb_transform(img.convert("RGB")) for img in image_sequence
                 ]
-                stacked_tensors = torch.stack(tensor_sequence, dim=0)
-                batch_images.append(stacked_tensors)
+                stacked_tensors_rgb = torch.stack(tensor_sequence_rgb, dim=0)
+                batch_rgb_tensors.append(stacked_tensors_rgb)
 
-            # Stack all batches to get shape (B, N, 3, H, W)
-            batch_images = torch.stack(batch_images, dim=0).to(device)
+                # Process Depth
+                if depth_images_pil[i] is not None: # Check if depth is actually present for this item
+                    depth_sequence = depth_images_pil[i]
+                    tensor_sequence_depth = [
+                        depth_transform(d_img.convert("L")) for d_img in depth_sequence
```

Comment:
We have to check this line, which will convert the depth image to 8-bit. I don't think it normalizes the depth image to the 0-1 range.

If this is the case, then we don't normalize images in the same way during inference and training, which is critical for performance.